src/meson/runners.ts

```diff
@@ -70,12 +70,14 @@ export async function runMesonReconfigure() {
 }
 
 export async function runMesonBuild(buildDir: string, name?: string) {
-  let command = !!name ? `ninja ${name}` : "ninja";
-  const stream = execStream(command, { cwd: buildDir });
+  const title = name ? `Building target ${name}` : "Building project";
+
+  getOutputChannel().append(`\n${title}\n`);
+  const stream = execStream(`meson compile ${name ?? ""}`, { cwd: buildDir });
```

Comment:
shouldn't this be quoted too.

Reply:
I'd only done test targets as surely nobody puts spaces in binary names 😀

That said, the one you reference ends up spawning and not going via the shell:

https://github.com/mesonbuild/vscode-meson/blob/fbede298913adb2f370466e159eb19f3ada7c842/src/utils.ts#L34-L42

So quoting breaks it there. The `FIXME` would solve the problem I suppose, though I get the feeling from looking at the code some greater refactoring to tidy up the multiple routes things can take is the right solution...

I'll update the PR for one other compile command that _is_ going via the shell for now.